Issue 25 model module

jhamacher-indeed opened the pull request.

jhamacher-indeed committed:
#25 Move files around.

jhamacher-indeed committed:
#25 Add build script for model.

jhamacher-indeed committed:
#25 Build operator as well.

jhamacher-indeed committed:
#25 Changes to enable internal publication of model module to Maven Central.

jhamacher-indeed committed:
#25 Move Dockerfile as well.

bfink13 commented on rabbitmq-operator-model/build.gradle:
```diff
@@ -0,0 +1,29 @@
+    mavenCentral()
+}
+
+dependencies {
```
i believe a whole bunch of these aren't needed

jhamacher-indeed replied:
I thought I'd verified that all of them were necessary, but it appears that I can remove jackson-databind and the kubernetes-client annotation processor.

I'll check again after moving some of the files out of the model module.

bfink13 reviewed (changes requested):
I think we only want the CRD models to move, right?

jhamacher-indeed commented:
Some of the files in model/crd/ depend on files in model/rabbitmq/.  For example:
```
package com.indeed.operators.rabbitmq.model.crd.rabbitmq;
...
import com.indeed.operators.rabbitmq.model.rabbitmq.VhostOperationPermissions;
...
public class VhostPermissions {
...
```

I didn't check extensively so maybe there are some non-CRD files that aren't necessary, but it seemed cleaner to move them all.

jhamacher-indeed committed:
Merge remote-tracking branch 'origin/master' into issue-25-model-module

bfink13 commented:
> Some of the files in model/crd/ depend on files in model/rabbitmq/. For example:

Oof, well that is extremely lame. I checked and it looks like VhostPermissions is the only file in the rabbitmq package that is referenced from the crd package. Can you just move VhostPermissions into the crd package instead? I'd rather not expose more than we have to in the public library.

jhamacher-indeed committed:
#25 More file shuffling.

We only want to publish the CRD-related files.  An additional class had to
be relocated to make this work.

jhamacher-indeed committed:
#25 Remove unnecessary build dependencies from the model.

jhamacher-indeed committed:
#25 Fix Travis build script.

jhamacher-indeed committed:
#25 Remove extra whitespace that snuck in.

bfink13 commented on rabbitmq-operator-model/build.gradle:
```diff
@@ -0,0 +1,23 @@
+    annotationProcessor('io.sundr:builder-annotations:0.14.7')
+}
+
+test {
```
this can be removed

bfink13 commented on rabbitmq-operator/build.gradle:
```diff
@@ -12,24 +12,20 @@ repositories {
     }
     compile('org.springframework.boot:spring-boot-starter-log4j2:2.1.1.RELEASE')
 
-    compileOnly('io.sundr:builder-annotations:0.14.7')
```
nice

bfink13 commented on release.sh:
```diff
@@ -1,7 +1,7 @@
 set -e
 
-docker build -t indeedoss/rabbitmq-operator:latest .
+docker build -t indeedoss/rabbitmq-operator:latest rabbitmq-operator
```
the Dockerfile should probably move into the rabbitmq-operator directory since that's all it builds against

jhamacher-indeed replied:
I did move the Dockerfile, back in 3e33903.  I just missed updating this until I saw the Travis test build fail.

jhamacher-indeed replied:
Or did you mean that this file should be moved into the rabbitmq-operator directory as well?  That would require changes to the Travis configuration (I think).

jhamacher-indeed committed:
#25 We don't have tests so remove it from the build.gradle.

jhamacher-indeed committed:
Merge remote-tracking branch 'origin/master' into issue-25-model-module

jhamacher-indeed commented:
@bfink13 any more feedback here?  I think I addressed everything.
